example/domain/lib/payments/usecase/initialize_single_payment_use_case.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:domain/kevin/model/payment.dart';
+import 'package:domain/kevin/model/payment_request.dart';
+import 'package:domain/kevin/repository/kevin_repository.dart';
+import 'package:domain/payments/model/payment_type.dart';
+
+class InitializeSinglePaymentUseCase {
+  final KevinRepository _kevinRepository;
+
+  const InitializeSinglePaymentUseCase({
+    required KevinRepository kevinRepository,
+  }) : _kevinRepository = kevinRepository;
+
+  Future<Payment> invoke({
+    required PaymentType paymentType,
+    required PaymentRequest paymentRequest,
+  }) {
+    switch (paymentType) {
+      case PaymentType.bank:
+        return _kevinRepository.initializeBankPayment(paymentRequest);
+      case PaymentType.card:
+        return _kevinRepository.initializeCardPayment(paymentRequest);
+      case PaymentType.linked:
+        throw StateError('Wrong payment type: $paymentType');
```

Comment:
I thought about having single use case, though that would require to rework `PaymentType` enum into abstract class, because `linked` type would have needed to have `linkedAccount` field. Otherwise, use case could have arguments `PaymentType type, LinkedAccount? account`, and I don't like an idea of having nullable account argument. So I believe that curreny option is the least evil.